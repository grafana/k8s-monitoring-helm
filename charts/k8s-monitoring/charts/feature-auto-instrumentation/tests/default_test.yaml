# yamllint disable rule:document-start rule:line-length rule:trailing-spaces
suite: Test default values
templates:
  - configmap.yaml
tests:
  - it: creates a module with default Beyla configuration
    set:
      deployAsConfigMap: true
    asserts:
      - isKind:
          of: ConfigMap
      - equal:
          path: data["module.alloy"]
          value: |-
            declare "auto_instrumentation" {
              argument "metrics_destinations" {
                comment = "Must be a list of metric destinations where collected metrics should be forwarded to"
              }
            
              discovery.kubernetes "beyla_pods" {
                role = "pod"
                namespaces {
                  own_namespace = true
                }
                selectors {
                  role = "pod"
                  label = "app.kubernetes.io/name=beyla"
                }
              }
          
              discovery.relabel "beyla_pods" {
                targets = discovery.kubernetes.beyla_pods.targets
                rule {
                  source_labels = ["__meta_kubernetes_pod_node_name"]
                  action = "replace"
                  target_label = "instance"
                }
              }
          
              prometheus.scrape "beyla_applications" {
                targets         = discovery.relabel.beyla_pods.output
                honor_labels    = true
                scrape_interval = "60s"
                clustering {
                  enabled = true
                }
                forward_to = argument.metrics_destinations.value
              }
          
              prometheus.scrape "beyla_internal" {
                targets         = discovery.relabel.beyla_pods.output
                metrics_path    = "/internal/metrics"
                job_name        = "integrations/beyla"
                honor_labels    = true
                scrape_interval = "60s"
                clustering {
                  enabled = true
                }
                forward_to = argument.metrics_destinations.value
              }
            }
