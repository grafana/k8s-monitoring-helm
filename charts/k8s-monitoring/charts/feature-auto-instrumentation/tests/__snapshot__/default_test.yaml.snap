creates a module with default Beyla configuration:
  1: |
    |-
      declare "auto_instrumentation" {
        argument "metrics_destinations" {
          comment = "Must be a list of metric destinations where collected metrics should be forwarded to"
        }

        discovery.kubernetes "beyla_pods" {
          role = "pod"
          namespaces {
            own_namespace = true
          }
          selectors {
            role = "pod"
            label = "app.kubernetes.io/name=beyla"
          }
        }

        discovery.relabel "beyla_pods" {
          targets = discovery.kubernetes.beyla_pods.targets
          rule {
            source_labels = ["__meta_kubernetes_pod_node_name"]
            action = "replace"
            target_label = "instance"
          }
        }

        prometheus.scrape "beyla_applications" {
          targets         = discovery.relabel.beyla_pods.output
          honor_labels    = true
          scrape_interval = "60s"
          clustering {
            enabled = true
          }
          forward_to = argument.metrics_destinations.value
        }

        prometheus.scrape "beyla_internal" {
          targets         = discovery.relabel.beyla_pods.output
          metrics_path    = "/internal/metrics"
          job_name        = "integrations/beyla"
          honor_labels    = true
          scrape_interval = "60s"
          clustering {
            enabled = true
          }
          forward_to = argument.metrics_destinations.value
        }
      }
